### PR TITLE
remove forward and back slashes from chunk filenames when formatting …

### PR DESCRIPTION
**Pull request by rbuels**
this PR fixes formatting of reference sequences when the names have forward slashes or backslashes in them.

references gitter convo:

image

### CODE REVIEW

**Comment by cmdcolin** [comment]
@rbuels we might need to do the same for gff

if I run on a gff

```
$ bin/flatfile-to-json.pl --gff test.gff --trackLabel testing
Unsuccessful stat on filename containing newline at /home/cdiesh/src/gmod/jbrowse/bin/../src/perl5/FeatureTrack.pm line 70.
Unsuccessful stat on filename containing newline at /home/cdiesh/src/gmod/jbrowse/bin/../src/perl5/JsonFileStorage.pm line 64.
```

test.gff
```
hi/I/have/slashes Swine Pathogen Analysis Resource (SPAR) mat_peptide     1    100   .       +       0       ID=GB000440/KU318406.1/A2MC2-P90/USA/2-5-2:nsp10;Name=nsp10;ORF1ab-gene;Note=full
```

**Comment by cmdcolin** [comment]
actually nevermind, seems to work :)

**Comment by rbuels** [comment]
maybe should escape newlines too, i'll add that

File: src/perl5/Bio/JBrowse/Cmd/FormatSequences.pm [hunk 1]
```diff
@@ -590,6 +590,9 @@ sub openChunkFile {
             "$refInfo->{name}-$chunkNum.txt"
           );
 
+
+    $file =~ s![/\\\n]!_!g; # convert filesystem reserved characters to underscore
```
Comment (cmdcolin) [inline src/perl5/Bio/JBrowse/Cmd/FormatSequences.pm:594]: why would there be a newline in the filename?

**Comment by btski** [comment]
Fix works great. Thanks!

**Comment by cmdcolin** [comment]
nice. we can go ahead and merge :)

**Comment by rbuels** [comment]
just thought of something, the JS code probably needs to be updated to match, right?

**Comment by cmdcolin** [comment]
looked ok to me without any additional changes, not sure what js you mean though?

**Comment by rbuels** [comment]
the JS code probably needs to do the same regex replace, in order to be able to find the munged paths for the chunk files

**Comment by cmdcolin** [comment]
didn't appear to be needed as far as I could tell, did this on this branch

    bin/prepare-refseqs.pl --fasta tests/data/forward_slashes_in_refseq_name.fa

and then opening up http://localhost/jbrowse worked fine without any additional changes

**Comment by cmdcolin** [comment]
[JBrowse-1.16.12-alpha.0.zip](https://github.com/GMOD/jbrowse/files/6206427/JBrowse-1.16.12-alpha.0.zip)

**Comment by cmdcolin** [comment]
I uploaded above @btski  a build that includes this fix plus the fixed refseq on dev just in case it helps :)

File: src/perl5/Bio/JBrowse/Cmd/FormatSequences.pm (hunk 1 above)
Comment (rbuels) [reply src/perl5/Bio/JBrowse/Cmd/FormatSequences.pm:594]: filename is made from the ref name, which might have a newline in it possibly

**Pull request merged**